### PR TITLE
chore: fix clippy warning for beta rust toolchain

### PR DESCRIPTION
**Pull request by MylesBorins**
/cc @devsnek @bengl I'm not sure that this actually makes things better PTAL

### CODE REVIEW

File: core/src/lib.rs [hunk 1]
```diff
@@ -26,7 +26,7 @@ pub fn init(v8_flags: Option<Vec<String>>) {
 /// Options for `JSTime::new`.
 pub struct Options {
     pub snapshot: Option<&'static [u8]>,
-    taking_snapshot: bool,
+    pub taking_snapshot: bool,
```
Comment (MylesBorins) [inline core/src/lib.rs:29]: This change was necessary to squelch a warning about field assignment after initializing struct using default

I'm not convinced this should have been made public, probably a better way to do this no?

```
warning: field assignment outside of initializer for an instance created with Default::default()
  --> cli/main.rs:36:5
   |
36 | /     options.snapshot = Some(include_bytes!(concat!(
37 | |         env!("OUT_DIR"),
38 | |         "/snapshot_data.blob"
39 | |     )));
   | |________^
   |
   = note: `#[warn(clippy::field_reassign_with_default)]` on by default
note: consider initializing the variable with `jstime_core::Options { snapshot: Some(include_bytes!(concat!(
        env!("OUT_DIR"),
        "/snapshot_data.blob"
    ))), ..Default::default() }` and removing relevant reassignments
  --> cli/main.rs:35:5
   |
35 |     let mut options = jstime::Options::default();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#field_reassign_with_default

warning: 1 warning emitted
```

Comment (devsnek) [reply core/src/lib.rs:29]: I would just add `#[allow(clippy::field_reassign_with_default)]` for the moment, the `taking_snapshot` field definitely shouldn't be public

Comment (MylesBorins) [reply core/src/lib.rs]: I updated code so we can use a `new` constructor to instantiate struct without exposing the internals

File: core/src/builtins/mod.rs [hunk 2]
```diff
@@ -51,7 +51,7 @@ impl Builtins {
 
 fn printer(scope: &mut v8::HandleScope, args: v8::FunctionCallbackArguments, _rv: v8::ReturnValue) {
     let arg_len = args.length();
-    assert!(arg_len >= 0 && arg_len <= 2);
+    assert!((0..2).contains(&arg_len));
```
Comment (devsnek) [inline core/src/builtins/mod.rs:54]: should this be `(0..=2)`?

Comment (MylesBorins) [reply core/src/builtins/mod.rs]: fixed

File: cli/main.rs [hunk 3]
```diff
@@ -32,11 +32,10 @@ fn main() {
             .map(|o| o.split(' ').map(|s| s.to_owned()).collect()),
     );
 
-    let mut options = jstime::Options::default();
-    options.snapshot = Some(include_bytes!(concat!(
+    let options = jstime::Options::new(Some(include_bytes!(concat!(
         env!("OUT_DIR"),
         "/snapshot_data.blob"
-    )));
+    ))));
```
Comment (MylesBorins) [inline cli/main.rs:38]: ))))))))))))))))))))))))))

**Comment by MylesBorins** [comment]
quick ping for reviews

**Review by bengl — APPROVED** [review]

**Pull request merged**